### DISCUSSION TITLE
Dark theme without boilerplate?

### DISCUSSION THREAD

**Discussion by 1Dragoon** [hecrj/iced#949] [Ideas]
Any chance we could get a built-in dark theme, thus avoiding the need for a lot of styling boilerplate code if we want to use it? Ideally implemented in such a way that it can be changed at runtime. Bonus points for having a means of auto-detecting how the browser is configured for web targets. (There's already a crate called dark-light that does this for windows/mac environments.)

**Comment by hecrj** [comment]
Maybe we could create a new `iced_themes` crate in the [`iced-rs`](https://github.com/iced-rs) organization?

If someone wants to take the lead, let me know!

**Reply by 13r0ck** [reply]
Iced themes would be cool. Long term I would like to look into that, no promises though. 

I think it would be usefull to make the styling system mostly rust macros so the developer only has to make small changes, and the rest is filled in by the compiler

Basically trying to make the styling more css like, but also with the compile time guarentees of rust 

🤷‍♂️. Just a thought

**Reply by cryptoquick** [reply]
I honestly prefer styles the way Iced does it. It took a bit of getting used to, though. Macros are sometimes kinda clunky in their own way, often not working well with tooling.

**Reply by cryptoquick** [reply]
This is how I did it in Fuzzr, btw. It's a little buggy with my latest changes, though. I need to get into these themes and widgets libraries, now that they're available.

https://github.com/FuzzrNet/Fuzzr/blob/34cdff07a48a906e3fc64191274779791015239d/src/ui/style.rs

**Reply by 90-008** [reply]
This is how I do theming on Crust (a chat app for a protocol) https://github.com/harmony-development/Crust/blob/master/src/style.rs It allows for "overriding" for finer grained theming. The app theming also supports reading colorschemes from a theme file, and it is reloaded on runtime automatically when the file is changed (https://github.com/harmony-development/Crust/blob/master/src/screen/mod.rs#L335 for sending file events, https://github.com/harmony-development/Crust/blob/master/src/screen/mod.rs#L614 for receiving)
